BUG:optimize: test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown fails

### Describe your issue.

I'm getting a test failure of 
`scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown`, as seen here:

```
$ SCIPY_XSLOW=1 spin test -m full -v -t scipy.optimize.tests.test_linprog::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
Invoking `build` prior to running tests:
$ meson compile -j 12 -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/warren/py3.14.3/bin/ninja -C /home/warren/repos/git/forks/scipy-fresh/build -j 12
ninja: Entering directory `/home/warren/repos/git/forks/scipy-fresh/build'
[2/356] Generating subprojects/highs/highs/HConfig.h with a custom command
$ meson install --only-changed -C build --destdir ../build-install --tags=runtime,python-runtime,tests,devel --skip-subprojects
$ export PYTHONPATH="/home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages"
$ /home/warren/py3.14.3/bin/python3.14 -P -c 'import scipy'
$ cd /home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages
$ /home/warren/py3.14.3/bin/python3.14 -P -m pytest -v --pyargs scipy.optimize.tests.test_linprog::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
========================================= test session starts ==========================================
platform linux -- Python 3.14.3, pytest-9.0.2, pluggy-1.6.0 -- /home/warren/py3.14.3/bin/python3.14
cachedir: .pytest_cache
hypothesis profile 'default'
rootdir: /home/warren/repos/git/forks/scipy-fresh
configfile: pytest.ini
plugins: xdist-3.8.0, scipy_doctest-2.2.0, cov-7.0.0, timeout-2.4.0, hypothesis-6.151.9
collected 1 item                                                                                       

scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown FAILED      [100%]

=============================================== FAILURES ===============================================
____________________________ TestLinprogHiGHSMIP.test_mip_rel_gap_passdown _____________________________

self = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x7dc4fcb3e250>

    @pytest.mark.xslow
    def test_mip_rel_gap_passdown(self):
        # MIP taken from test_mip6, solved with different values of mip_rel_gap
        # solve a larger MIP with only equality constraints
        # source: https://www.mathworks.com/help/optim/ug/intlinprog.html
        A_eq = np.array([[22, 13, 26, 33, 21, 3, 14, 26],
                         [39, 16, 22, 28, 26, 30, 23, 24],
                         [18, 14, 29, 27, 30, 38, 26, 26],
                         [41, 26, 28, 36, 18, 38, 16, 26]])
        b_eq = np.array([7872, 10466, 11322, 12058])
        c = np.array([2, 10, 13, 17, 7, 5, 7, 3])
    
        bounds = [(0, np.inf)]*8
        integrality = [1]*8
    
        mip_rel_gaps = [0.5, 0.25, 0.01, 0.001]
        sol_mip_gaps = []
        for mip_rel_gap in mip_rel_gaps:
            res = linprog(c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
                          bounds=bounds, method=self.method,
                          integrality=integrality,
                          options={"mip_rel_gap": mip_rel_gap})
            final_mip_gap = res["mip_gap"]
            # assert that the solution actually has mip_gap lower than the
            # required mip_rel_gap supplied
            assert final_mip_gap <= mip_rel_gap
            sol_mip_gaps.append(final_mip_gap)
    
        # make sure that the mip_rel_gap parameter is actually doing something
        # check that differences between solution gaps are declining
        # monotonically with the mip_rel_gap parameter. np.diff does
        # x[i+1] - x[i], so flip the array before differencing to get
        # what should be a positive, monotone decreasing series of solution
        # gaps
        gap_diffs = np.diff(np.flip(sol_mip_gaps))
>       assert np.all(gap_diffs >= 0)
E       assert np.False_
E        +  where np.False_ = <function all at 0x7dc5af339130>(array([ 9.16936354e-03,  2.75080906e-02, -2.72004641e-15]) >= 0)
E        +    where <function all at 0x7dc5af339130> = np.all

A_eq       = array([[22, 13, 26, 33, 21,  3, 14, 26],
       [39, 16, 22, 28, 26, 30, 23, 24],
       [18, 14, 29, 27, 30, 38, 26, 26],
       [41, 26, 28, 36, 18, 38, 16, 26]])
b_eq       = array([ 7872, 10466, 11322, 12058])
bounds     = [(0, inf), (0, inf), (0, inf), (0, inf), (0, inf), (0, inf), ...]
c          = array([ 2, 10, 13, 17,  7,  5,  7,  3])
final_mip_gap = 0.0005393743257824605
gap_diffs  = array([ 9.16936354e-03,  2.75080906e-02, -2.72004641e-15])
integrality = [1, 1, 1, 1, 1, 1, ...]
mip_rel_gap = 0.001
mip_rel_gaps = [0.5, 0.25, 0.01, 0.001]
res        =         message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status...[]
                 marginals: []
 mip_node_count: 28157
 mip_dual_bound: 1853.0
        mip_gap: 0.0005393743257824605
self       = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x7dc4fcb3e250>
sol_mip_gaps = [0.037216828478961565, 0.037216828478964285, 0.009708737864078035, 0.0005393743257824605]

scipy/optimize/tests/test_linprog.py:2511: AssertionError
======================================= short test summary info ========================================
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown - assert np.False_
========================================== 1 failed in 11.08s ==========================================

```

<details>
<summary>SciPy/NumPy/Python version and system information</summary>

```shell
1.18.0.dev0+git20260313.dd4723b 2.4.3 sys.version_info(major=3, minor=14, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 3.0.2
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.2.4
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.14.3/lib/python3.14/site-packages/pythran
    version: 0.18.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.14.3/bin/python3.14
  version: '3.14'
```

</details>

I'm not convinced that the failing test is actually a valid test.  In the test [`test_mip_rel_passown()`](https://github.com/scipy/scipy/blob/b2ac0886cfab5759d8f06bffac1993d7b6710894/scipy/optimize/tests/test_linprog.py#L2477), I modified this line:
https://github.com/scipy/scipy/blob/b2ac0886cfab5759d8f06bffac1993d7b6710894/scipy/optimize/tests/test_linprog.py#L2491

to
```
        mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
```

and ran

```
$ SCIPY_XSLOW=1 spin test -m full -v -t scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
```

As before, the test fails the check that `sol_mip_gaps` is monotonically decreasing:

<details>
<summary>Test output</summary>

```
Invoking `build` prior to running tests:
$ meson compile -j 12 -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/warren/py3.14.5/bin/ninja -C /home/warren/repos/git/forks/scipy-fresh/build -j 12
ninja: Entering directory `/home/warren/repos/git/forks/scipy-fresh/build'
[2/356] Generating subprojects/highs/highs/HConfig.h with a custom command
$ meson install --no-rebuild --only-changed -C build --destdir ../build-install --tags=runtime,python-runtime,tests,devel --skip-subprojects
$ export PYTHONPATH="/home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages"
$ /home/warren/py3.14.5/bin/python3.14 -P -c 'import scipy'
$ cd /home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages
$ /home/warren/py3.14.5/bin/python3.14 -P -m pytest -v scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
======================================================== test session starts ========================================================
platform linux -- Python 3.14.5, pytest-9.0.3, pluggy-1.6.0 -- /home/warren/py3.14.5/bin/python3.14
cachedir: .pytest_cache
hypothesis profile 'deterministic' -> database=None, deadline=None, print_blob=True, derandomize=True, suppress_health_check=(HealthCheck.data_too_large, HealthCheck.filter_too_much, HealthCheck.too_slow, HealthCheck.large_base_example, HealthCheck.function_scoped_fixture, HealthCheck.differing_executors, HealthCheck.nested_given)
rootdir: /home/warren/repos/git/forks/scipy-fresh
configfile: pytest.ini
plugins: cov-7.1.0, anyio-4.13.0, timeout-2.4.0, hypothesis-6.152.9, xdist-3.8.0
collected 1 item

scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown FAILED                                   [100%]

============================================================= FAILURES ==============================================================
___________________________________________ TestLinprogHiGHSMIP.test_mip_rel_gap_passdown ___________________________________________

self = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x76211514a7a0>

    @pytest.mark.xslow
    def test_mip_rel_gap_passdown(self):
        # MIP taken from test_mip6, solved with different values of mip_rel_gap
        # solve a larger MIP with only equality constraints
        # source: https://www.mathworks.com/help/optim/ug/intlinprog.html
        A_eq = np.array([[22, 13, 26, 33, 21, 3, 14, 26],
                         [39, 16, 22, 28, 26, 30, 23, 24],
                         [18, 14, 29, 27, 30, 38, 26, 26],
                         [41, 26, 28, 36, 18, 38, 16, 26]])
        b_eq = np.array([7872, 10466, 11322, 12058])
        c = np.array([2, 10, 13, 17, 7, 5, 7, 3])

        bounds = [(0, np.inf)]*8
        integrality = [1]*8

        # mip_rel_gaps = [0.5, 0.25, 0.01, 0.001]
        mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
        sol_mip_gaps = []
        for mip_rel_gap in mip_rel_gaps:
            res = linprog(c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
                          bounds=bounds, method=self.method,
                          integrality=integrality,
                          options={"mip_rel_gap": mip_rel_gap})
            final_mip_gap = res["mip_gap"]
            # assert that the solution actually has mip_gap lower than the
            # required mip_rel_gap supplied
            assert final_mip_gap <= mip_rel_gap
            sol_mip_gaps.append(final_mip_gap)

        # make sure that the mip_rel_gap parameter is actually doing something
        # check that differences between solution gaps are declining
        # monotonically with the mip_rel_gap parameter. np.diff does
        # x[i+1] - x[i], so flip the array before differencing to get
        # what should be a positive, monotone decreasing series of solution
        # gaps
        gap_diffs = np.diff(np.flip(sol_mip_gaps))
>       assert np.all(gap_diffs >= 0)
E       assert np.False_
E        +  where np.False_ = <function all at 0x7621a3516470>(array([ 1.08961126e-01, -1.08961126e-01, -3.77562028e-03,  2.39718854e-01,\n        1.11022302e-16]) >= 0)
E        +    where <function all at 0x7621a3516470> = np.all

A_eq       = array([[22, 13, 26, 33, 21,  3, 14, 26],
       [39, 16, 22, 28, 26, 30, 23, 24],
       [18, 14, 29, 27, 30, 38, 26, 26],
       [41, 26, 28, 36, 18, 38, 16, 26]])
b_eq       = array([ 7872, 10466, 11322, 12058])
bounds     = [(0, inf), (0, inf), (0, inf), (0, inf), (0, inf), (0, inf), ...]
c          = array([ 2, 10, 13, 17,  7,  5,  7,  3])
final_mip_gap = 0.0409924487594394
gap_diffs  = array([ 1.08961126e-01, -1.08961126e-01, -3.77562028e-03,  2.39718854e-01,
        1.11022302e-16])
integrality = [1, 1, 1, 1, 1, 1, ...]
mip_rel_gap = 0.1
mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
res        =         message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status...l: []
                 marginals: []
 mip_node_count: 18446
 mip_dual_bound: 1778.0
        mip_gap: 0.0409924487594394
self       = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x76211514a7a0>
sol_mip_gaps = [0.27693568209750075, 0.27693568209750064, 0.037216828478964285, 0.0409924487594387, 0.14995357474466037, 0.0409924487594394]

scipy/optimize/tests/test_linprog.py:2512: AssertionError
====================================================== short test summary info ======================================================
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown - assert np.False_
======================================================== 1 failed in 14.85s =========================================================
```

</details>

From the log, the computed gaps are

```
sol_mip_gaps = [0.27693568209750075, 0.27693568209750064, 0.037216828478964285, 0.0409924487594387, 0.14995357474466037, 0.0409924487594394]
```

Each of these gaps is less than the corresponding value in `min_rel_gaps`.  I don't see why these values should be monotonically decreasing. The results show that in some cases the computed gap is a lot less than the requested value, and other in other cases the gap is just barely less than the requested gap (as in the value `0.15` above, where the corresponding value in `sol_mip_gaps` is `0.14995357474466037`).  The expectation that `sol_min_gaps` is monotonically decreasing seems incorrect.

If my understanding is correct, we should just remove this part of the test.

> I'm not convinced that the failing test is actually a valid test. In the test [`test_mip_rel_passown()`](https://github.com/scipy/scipy/blob/b2ac0886cfab5759d8f06bffac1993d7b6710894/scipy/optimize/tests/test_linprog.py#L2477), I modified this line:
> 
> [scipy/scipy/optimize/tests/test_linprog.py](https://github.com/scipy/scipy/blob/b2ac0886cfab5759d8f06bffac1993d7b6710894/scipy/optimize/tests/test_linprog.py#L2491)
> 
> Line 2491 in [b2ac088](/scipy/scipy/commit/b2ac0886cfab5759d8f06bffac1993d7b6710894)
> 
>  mip_rel_gaps = [0.5, 0.25, 0.01, 0.001] 
> to
> 
> ```
>         mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
> ```
> 
> and ran
> 
> ```
> $ SCIPY_XSLOW=1 spin test -m full -v -t scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
> ```
> 
> As before, the test fails the check that `sol_mip_gaps` is monotonically decreasing:
> 
> Test output
> ```
> Invoking `build` prior to running tests:
> $ meson compile -j 12 -C build
> INFO: autodetecting backend as ninja
> INFO: calculating backend command to run: /home/warren/py3.14.5/bin/ninja -C /home/warren/repos/git/forks/scipy-fresh/build -j 12
> ninja: Entering directory `/home/warren/repos/git/forks/scipy-fresh/build'
> [2/356] Generating subprojects/highs/highs/HConfig.h with a custom command
> $ meson install --no-rebuild --only-changed -C build --destdir ../build-install --tags=runtime,python-runtime,tests,devel --skip-subprojects
> $ export PYTHONPATH="/home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages"
> $ /home/warren/py3.14.5/bin/python3.14 -P -c 'import scipy'
> $ cd /home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.14/site-packages
> $ /home/warren/py3.14.5/bin/python3.14 -P -m pytest -v scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown
> ======================================================== test session starts ========================================================
> platform linux -- Python 3.14.5, pytest-9.0.3, pluggy-1.6.0 -- /home/warren/py3.14.5/bin/python3.14
> cachedir: .pytest_cache
> hypothesis profile 'deterministic' -> database=None, deadline=None, print_blob=True, derandomize=True, suppress_health_check=(HealthCheck.data_too_large, HealthCheck.filter_too_much, HealthCheck.too_slow, HealthCheck.large_base_example, HealthCheck.function_scoped_fixture, HealthCheck.differing_executors, HealthCheck.nested_given)
> rootdir: /home/warren/repos/git/forks/scipy-fresh
> configfile: pytest.ini
> plugins: cov-7.1.0, anyio-4.13.0, timeout-2.4.0, hypothesis-6.152.9, xdist-3.8.0
> collected 1 item
> 
> scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown FAILED                                   [100%]
> 
> ============================================================= FAILURES ==============================================================
> ___________________________________________ TestLinprogHiGHSMIP.test_mip_rel_gap_passdown ___________________________________________
> 
> self = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x76211514a7a0>
> 
>     @pytest.mark.xslow
>     def test_mip_rel_gap_passdown(self):
>         # MIP taken from test_mip6, solved with different values of mip_rel_gap
>         # solve a larger MIP with only equality constraints
>         # source: https://www.mathworks.com/help/optim/ug/intlinprog.html
>         A_eq = np.array([[22, 13, 26, 33, 21, 3, 14, 26],
>                          [39, 16, 22, 28, 26, 30, 23, 24],
>                          [18, 14, 29, 27, 30, 38, 26, 26],
>                          [41, 26, 28, 36, 18, 38, 16, 26]])
>         b_eq = np.array([7872, 10466, 11322, 12058])
>         c = np.array([2, 10, 13, 17, 7, 5, 7, 3])
> 
>         bounds = [(0, np.inf)]*8
>         integrality = [1]*8
> 
>         # mip_rel_gaps = [0.5, 0.25, 0.01, 0.001]
>         mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
>         sol_mip_gaps = []
>         for mip_rel_gap in mip_rel_gaps:
>             res = linprog(c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
>                           bounds=bounds, method=self.method,
>                           integrality=integrality,
>                           options={"mip_rel_gap": mip_rel_gap})
>             final_mip_gap = res["mip_gap"]
>             # assert that the solution actually has mip_gap lower than the
>             # required mip_rel_gap supplied
>             assert final_mip_gap <= mip_rel_gap
>             sol_mip_gaps.append(final_mip_gap)
> 
>         # make sure that the mip_rel_gap parameter is actually doing something
>         # check that differences between solution gaps are declining
>         # monotonically with the mip_rel_gap parameter. np.diff does
>         # x[i+1] - x[i], so flip the array before differencing to get
>         # what should be a positive, monotone decreasing series of solution
>         # gaps
>         gap_diffs = np.diff(np.flip(sol_mip_gaps))
> >       assert np.all(gap_diffs >= 0)
> E       assert np.False_
> E        +  where np.False_ = <function all at 0x7621a3516470>(array([ 1.08961126e-01, -1.08961126e-01, -3.77562028e-03,  2.39718854e-01,\n        1.11022302e-16]) >= 0)
> E        +    where <function all at 0x7621a3516470> = np.all
> 
> A_eq       = array([[22, 13, 26, 33, 21,  3, 14, 26],
>        [39, 16, 22, 28, 26, 30, 23, 24],
>        [18, 14, 29, 27, 30, 38, 26, 26],
>        [41, 26, 28, 36, 18, 38, 16, 26]])
> b_eq       = array([ 7872, 10466, 11322, 12058])
> bounds     = [(0, inf), (0, inf), (0, inf), (0, inf), (0, inf), (0, inf), ...]
> c          = array([ 2, 10, 13, 17,  7,  5,  7,  3])
> final_mip_gap = 0.0409924487594394
> gap_diffs  = array([ 1.08961126e-01, -1.08961126e-01, -3.77562028e-03,  2.39718854e-01,
>         1.11022302e-16])
> integrality = [1, 1, 1, 1, 1, 1, ...]
> mip_rel_gap = 0.1
> mip_rel_gaps = [0.35, 0.3, 0.25, 0.2, 0.15, 0.1]
> res        =         message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
>         success: True
>          status...l: []
>                  marginals: []
>  mip_node_count: 18446
>  mip_dual_bound: 1778.0
>         mip_gap: 0.0409924487594394
> self       = <scipy.optimize.tests.test_linprog.TestLinprogHiGHSMIP object at 0x76211514a7a0>
> sol_mip_gaps = [0.27693568209750075, 0.27693568209750064, 0.037216828478964285, 0.0409924487594387, 0.14995357474466037, 0.0409924487594394]
> 
> scipy/optimize/tests/test_linprog.py:2512: AssertionError
> ====================================================== short test summary info ======================================================
> FAILED scipy/optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip_rel_gap_passdown - assert np.False_
> ======================================================== 1 failed in 14.85s =========================================================
> ```
> 
> From the log, the computed gaps are
> 
> ```
> sol_mip_gaps = [0.27693568209750075, 0.27693568209750064, 0.037216828478964285, 0.0409924487594387, 0.14995357474466037, 0.0409924487594394]
> ```
> 
> Each of these gaps is less than the corresponding value in `min_rel_gaps`. I don't see why these values should be monotonically decreasing. The results show that in some cases the computed gap is a lot less than the requested value, and other in other cases the gap is just barely less than the requested gap (as in the value `0.15` above, where the corresponding value in `sol_mip_gaps` is `0.14995357474466037`). The expectation that `sol_min_gaps` is monotonically decreasing seems incorrect.
> 
> If my understanding is correct, we should just remove this part of the test.

@amlimaye as the original test author do you have a moment to comment on this?

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -2497,6 +2497,16 @@ class TestLinprogHiGHSMIP:
         # required mip_rel_gap supplied
         assert final_mip_gap <= mip_rel_gap
 
+        msg = ("The second part of this test requires final_mip_gap > 0.  If "
+               "final_mip_gap == 0, we can't test that rerunning linprog() "
+               "with mip_rel_gap set to 0.999*final_mip_gap results in a smaller "
+               "computed mip_gap.  To fix this, a new test problem is needed for "
+               "which the computed mip_gap is not zero.  This test failuure does "
+               "not indicate that there is bug in HiGHS or in the SciPy wrapper. "
+               "It just means that the behavior of HiGHS on the test problem has "
+               "changed in a way that makes the second part of the test pointless.")
+        assert final_mip_gap > 0, msg
+
         # Do it again, feeding a slight decreased final_mip_gap back as the
         # mip_rel_gap option, to ensure that we see the effect of the option.
         smaller_mip_rel_gap = 0.999 * final_mip_gap

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -2474,7 +2474,8 @@ class TestLinprogHiGHSMIP:
         np.testing.assert_allclose(res.fun, 1854)
 
     @pytest.mark.xslow
-    def test_mip_rel_gap_passdown(self):
+    @pytest.mark.parametrize('mip_rel_gap', [0.5, 0.001])
+    def test_mip_rel_gap_passdown(self, mip_rel_gap):
         # MIP taken from test_mip6, solved with different values of mip_rel_gap
         # solve a larger MIP with only equality constraints
         # source: https://www.mathworks.com/help/optim/ug/intlinprog.html
@@ -2487,29 +2488,24 @@ class TestLinprogHiGHSMIP:
 
         bounds = [(0, np.inf)]*8
         integrality = [1]*8
+        res = linprog(c=c, A_ub=None, b_ub=None, A_eq=A_eq, b_eq=b_eq,
+                      bounds=bounds, method=self.method,
+                      integrality=integrality,
+                      options={"mip_rel_gap": mip_rel_gap})
+        final_mip_gap = res["mip_gap"]
+        # assert that the solution actually has mip_gap lower than the
+        # required mip_rel_gap supplied
+        assert final_mip_gap <= mip_rel_gap
 
-        mip_rel_gaps = [0.5, 0.25, 0.01, 0.001]
-        sol_mip_gaps = []
-        for mip_rel_gap in mip_rel_gaps:
-            res = linprog(c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
-                          bounds=bounds, method=self.method,
-                          integrality=integrality,
-                          options={"mip_rel_gap": mip_rel_gap})
-            final_mip_gap = res["mip_gap"]
-            # assert that the solution actually has mip_gap lower than the
-            # required mip_rel_gap supplied
-            assert final_mip_gap <= mip_rel_gap
-            sol_mip_gaps.append(final_mip_gap)
-
-        # make sure that the mip_rel_gap parameter is actually doing something
-        # check that differences between solution gaps are declining
-        # monotonically with the mip_rel_gap parameter. np.diff does
-        # x[i+1] - x[i], so flip the array before differencing to get
-        # what should be a positive, monotone decreasing series of solution
-        # gaps
-        gap_diffs = np.diff(np.flip(sol_mip_gaps))
-        assert np.all(gap_diffs >= 0)
-        assert not np.all(gap_diffs == 0)
+        # Do it again, feeding a slight decreased final_mip_gap back as the
+        # mip_rel_gap option, to ensure that we see the effect of the option.
+        smaller_mip_rel_gap = 0.999 * final_mip_gap
+        res = linprog(c=c, A_ub=None, b_ub=None, A_eq=A_eq, b_eq=b_eq,
+                      bounds=bounds, method=self.method,
+                      integrality=integrality,
+                      options={"mip_rel_gap": smaller_mip_rel_gap})
+        smaller_final_mip_gap = res["mip_gap"]
+        assert smaller_final_mip_gap <= smaller_mip_rel_gap
 
     def test_semi_continuous(self):
         # See issue #18106. This tests whether the solution is being

